graph/schema.resolvers.go

```
@@ -0,0 +1,141 @@
+	var pa []*model.PokemonAbility
+
+	for _, ability_input := range pkmn_ability_input {
+		a := model.PokemonAbility{
+			Name: ability_input.Name,
+		}
+		pa = append(pa, &a)
+	}
+	return pa
+}
+
+// CreatePokemon is the resolver for the createPokemon field.
+func (r *mutationResolver) CreatePokemon(ctx context.Context, input model.PokemonInput) (*model.Pokemon, error) {
+	//r.DB.Select()
+	//fmt.Println(r.DB)
+
+	new_pkmn := model.Pokemon{
+		Name:        input.Name,
+		Description: input.Description,
+		Category:    input.Category,
+		Type:        TypeInputMap(input.Type),
+		Abilities:   AbilityInputMap(input.Abilities),
+	}
+
+	//add to database
+	err := r.DB.Create(&new_pkmn).Error
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("hereeeee")
+	fmt.Println(new_pkmn.Name)
+
+	return &new_pkmn, nil
+	//panic(fmt.Errorf("not implemented: CreatePokemon - createPokemon"))
+}
+
+// UpdatePokemon is the resolver for the updatePokemon field.
+func (r *mutationResolver) UpdatePokemon(ctx context.Context, id int, input model.PokemonInput) (*model.Pokemon, error) {
+	//first check if ID exists or not
+	var poke model.Pokemon
+	p := r.DB.First(&poke, id)
+	if p.Error != nil {
+		return nil, p.Error
+	}
+
+	updated_poke := model.Pokemon{
+		ID:          id,
+		Name:        input.Name,
+		Description: input.Description,
+		Category:    input.Category,
+		Type:        TypeInputMap(input.Type),
+		Abilities:   AbilityInputMap(input.Abilities),
+	}
+	r.DB.Save(&updated_poke)
+	return &updated_poke, nil
+	//panic(fmt.Errorf("not implemented: UpdatePokemon - updatePokemon"))
+}
+
+// DeletePokemon is the resolver for the deletePokemon field.
+func (r *mutationResolver) DeletePokemon(ctx context.Context, id int) (bool, error) {
+	p := r.DB.Delete(&model.Pokemon{}, id)
+	if p.RowsAffected < 1 {
+		fmt.Printf("numbers of row affected: %v\n", p.RowsAffected)
+		fmt.Println("The provided id doesn't exist")
+		return false, nil //FIXME: nil here is a placeholder
```

should return an error